yktrace/src/tir.rs

```diff
@@ -229,15 +262,29 @@ impl TirTrace {
                 e => panic!("Expected `StorageLive` here, instead got {:?}.", e)
             }
         }
+        let mut removed_dead = false;
```

Comment:
In the future we could do with a less fragile way of killing this stuff. Did you have any ideas? I know you said that tracing certain types of code makes this panic? Was it code without a call?

Reply:
To be honest I can't quite remember. This definitely needs rewriting although I don't yet have a good idea how.

Reply:
Might be worth having a thing about it, as I think this will bite us. Not for this PR obviously.